pallets/creator-staking/src/inflation.rs

```diff
@@ -0,0 +1,44 @@
+use crate::*;
+use frame_support::traits::{Currency, Get, OnTimestampSet, Imbalance};
+use sp_runtime::traits::{Saturating, SaturatedConversion, UniqueSaturatedInto, Zero};
+
+type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+impl<Moment, T: Config> OnTimestampSet<Moment> for Pallet<T> {
+    fn on_timestamp_set(_moment: Moment) {
+        let total_issuance = T::Currency::total_issuance();
+        let reward_amount: BalanceOf<T> = T::CurrentAnnualInflation::get() * total_issuance
+            / T::BlocksPerYear::get().saturated_into::<u32>().unique_saturated_into();
+
+        let inflation = T::Currency::issue(reward_amount);
+        Self::distribute_rewards(inflation);
```

Comment:
Why do we use a returned value from `Currency::issue()` instead of `reward_amount`

Reply:
Because `Currency:issue` doesn’t guarantee that the entire amount is issued (e.g. in case of overflow).
ref: https://paritytech.github.io/substrate/master/frame_support/traits/tokens/currency/trait.Currency.html#tymethod.issue

Reply:
> Because `Currency:issue` doesn’t guarantee that the entire amount is issued (e.g. in case of overflow). ref: https://paritytech.github.io/substrate/master/frame_support/traits/tokens/currency/trait.Currency.html#tymethod.issue

Just make it as a comment before this line of code :)